File: python-sdk/src/boxsdk/models/operations/post_sign_requests.py
import dataclasses
from typing import Optional
from boxsdk import utils
from ..shared import signrequestcreaterequest as shared_signrequestcreaterequest
from ..shared import clienterror as shared_clienterror
from ..shared import signrequest as shared_signrequest


@dataclasses.dataclass
class PostSignRequestsRequest:
    request: Optional[shared_signrequestcreaterequest.SignRequestCreateRequest] = dataclasses.field(default=None, metadata={'request': { 'media_type': 'application/json' }})
    retries: Optional[utils.RetryConfig] = dataclasses.field(default=None)
    

@dataclasses.dataclass
class PostSignRequestsResponseOutput:
    content_type: str = dataclasses.field()
    status_code: int = dataclasses.field()
    client_error: Optional[shared_clienterror.ClientError] = dataclasses.field(default=None)
    sign_request: Optional[shared_signrequest.SignRequestOutput] = dataclasses.field(default=None)
    

File: sdks/ts/src/sdk/models/operations/postsignrequests.ts
import { SpeakeasyMetadata, SpeakeasyBase, RetryConfig } from "../../../internal/utils";
import * as shared from "../shared";



export class PostSignRequestsRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  retries?: RetryConfig;

  @SpeakeasyMetadata({ data: "request, media_type=application/json" })
  request?: shared.SignRequestCreateRequest;
}


export class PostSignRequestsResponseOutput extends SpeakeasyBase {
  @SpeakeasyMetadata()
  clientError?: shared.ClientError;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  signRequest?: shared.SignRequestOutput;

  @SpeakeasyMetadata()
  statusCode: number;
}
